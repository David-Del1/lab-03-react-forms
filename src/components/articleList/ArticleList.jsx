import React from 'react';
import PropTypes from 'prop-types';
import Article from '../article/Article';


const ArticleList = ({ articles }) => {
  const listItems = articles.map(article => (
    <li
      key={article.title}
    >
      <Article article={article}/>
    </li>
  ));
  return <ul>{listItems}</ul>;
};

ArticleList.propTypes = {
  articles: PropTypes.arrayOf(
    PropTypes.shape({
      title: PropTypes.string.isRequired,
      author: PropTypes.string.isRequired,
      description: PropTypes.string.isRequired
    })
  )
};

export default ArticleList;
