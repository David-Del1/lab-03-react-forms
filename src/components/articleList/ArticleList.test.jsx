import React from 'react';
import { render } from '@testing-library/react';
import ArticleList from './ArticleList.jsx';

describe('Article list', () => {
  it('renders a title, author, and description', () => {
    const articles = [{
      title: 'Lorem ipsum',
      author: 'lorem',
      description: 'lorem'
    },
    {
      title: 'lorem2',
      author: 'ipsum',
      description: 'hello'
    }];

    const { asFragment } = render(<ArticleList articles={articles} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
