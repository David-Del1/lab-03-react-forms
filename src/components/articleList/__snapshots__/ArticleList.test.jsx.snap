// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article list renders a title, author, and description 1`] = `
<DocumentFragment>
  <ul>
    <li>
      <div>
        <h1>
          Lorem ipsum
        </h1>
        <p>
          lorem
        </p>
        <p>
           lorem
        </p>
      </div>
    </li>
    <li>
      <div>
        <h1>
          lorem2
        </h1>
        <p>
          ipsum
        </p>
        <p>
           hello
        </p>
      </div>
    </li>
  </ul>
</DocumentFragment>
`;
